### ISSUE TITLE
adopt didcomm:transport/queue

### ISSUE THREAD

**Issue by TimoGlastra** [hyperledger/aries-cloudagent-python#1144]
ACA-Py currently expects an empty string when agents without an endpoint connect. It is never formally defined in Aries, but both  AFJ and AFGO use `didcomm:transport/queue`. 

It was previously discussed in an Aries issue: https://github.com/hyperledger/aries-rfcs/issues/405 and is also the standard for DIDComm v2: https://github.com/decentralized-identity/didcomm-messaging/blob/bd2e36ffb31a60da417d57f7aecff2c89b7b7f95/extensions/return_route/main.md#queue-transport.

I think .NET doesn't use this, but we could just accept both an empty string and `didcomm:transport/queue`.

Thoughts?

**Comment by swcurran** [comment]
Sounds good -- @shaangill025 can you add this to your list.  Perhaps check with @TimoGlastra if necessary.

**Comment by TimoGlastra** [comment]
After a bit more investigation it seems like it doesn't matter to ACA-Py what endpoint is used. It checks whether it has a supported transport and otherwise puts it in the "waiting for pickup" bucket.
So could be empty string, `didcomm:transport/queue` or any other string

**Comment by marcinkiewiczblazej** [comment]
Currently I'm trying to setup mediator using ACA-Py (with `aries-acapy-plugin-toolbox`) with AFJ react native. And it seems that the issue is still open. When negotiating connection I see this line in logs:

```
aries_cloudagent.core.conductor WARNING Cannot queue message for delivery, no supported transport
```

What happens next is that on the React native side I'm getting a timeout as the response is not delivered from mediator. @TimoGlastra is there a way to pass the config so that `didcomm:transport/queue` are delivered?

My current config:
```
plugin:
    - acapy_plugin_toolbox

# General
label: Mindy Mediator
endpoint: http://endpoint:3001
inbound-transport:
  - [acapy_plugin_toolbox.http_ws, 0.0.0.0, 3001]
outbound-transport:
  - http
log-level: DEBUG

# Mediator does not use ledger
no-ledger: true

# Admin
admin: [0.0.0.0, 3005]
admin-insecure-mode: true

# Connections
debug-connections: true
connections-invite: true
invite-multi-use: true
auto-accept-invites: true
auto-accept-requests: true
auto-ping-connection: true

# Mediation
open-mediation: true
enable-undelivered-queue: true
```

**Comment by TimoGlastra** [comment]
Could it be related to this issue? You don't need the didcomm transport queue support in ACA-Py, it should work without.

@dbluhm does this sound familiar with the issue you were running into?

https://github.com/hyperledger/aries-cloudagent-python/pull/1469#issuecomment-984233912

**Comment by dbluhm** [comment]
Yes, this sounds like the issue we were experiencing. @marcinkiewiczblazej is this on ACA-Py version 0.7.2?

**Comment by swcurran** [comment]
I just merged #1469 .  Could you please try again with the current main branch?

**Comment by marcinkiewiczblazej** [comment]
I was checking out main repository few days ago, so no specific version. I'm running M1 Mac, so there are some additional problems I'm resolving on the way most of them are caused by no binary distributions for `linux/arm64`. So I have to build them from scratch.

Today I've tested it with latest main branch. And it helped a bit. So now messages are delivered, but I'm getting an error on  a request on the mobile client.

```
ERROR: Message is undeliverable to connection ... "@type": "https://didcomm.org/coordinate-mediation/1.0/mediate-request" ...
```

I've had problems with aries-askar and few other dependencies not available for M1. So there's a chance that something is not built properly yet. I'll be trying to fix it this week and get back to you with result.
